metaflow/metaflow_config.py

```diff
@@ -198,4 +199,56 @@ def get_authenticated_boto3_client(module, params={}):
             except requests.exceptions.HTTPError as e:
                 raise MetaflowException(repr(e))
         return boto3.session.Session(**cached_aws_sandbox_creds).client(module, **params)
+
+    if METAFLOW_AWS_ARN:
+        import logging
+        from datetime import datetime
+
+        import boto3
+        from botocore.credentials import AssumeRoleCredentialFetcher, DeferredRefreshableCredentials
+        from botocore.session import Session
+        from dateutil.tz import tzlocal
+
+        _logger = logging.getLogger(__name__)
+
+        def get_aws_session(role_arn: str) -> Session:
+            """
+            Args:
+                role_arn: AWS ARN
+            Returns: boto3 Session
+            """
+            source_session = boto3.Session()
+
+            # Use profile to fetch assume role credentials
+            fetcher = AssumeRoleCredentialFetcher(
+                client_creator=source_session._session.create_client,
+                source_credentials=source_session.get_credentials(),
+                role_arn=role_arn,
+            )
+
+            # Create new session with assumed role and auto-refresh
+            botocore_session = Session()
+            botocore_session._credentials = DeferredRefreshableCredentials(
+                method="assume-role",
+                refresh_using=fetcher.fetch_credentials,
+                time_fetcher=lambda: datetime.now(tzlocal()),
+            )
+
+            return botocore_session
+
+        def get_aws_client(role_arn: str, service: str):
+            """
+            Args:s
+                role_arn: AWS ARN
+                service: AWS service, example: "s3"
+
+            Returns: Returns an AWS client
+
```

Comment:
nit: Unneeded extra newline

Reply:
I've modified this code a bit to make it more readable and removed these parts that weren't necessary